main-linux.cpp

```diff
@@ -122,17 +122,102 @@ assert_handler (const jerry_api_object_t *function_obj_p __attr_unused___, /** <
                 const jerry_api_value_t args_p[], /** < function arguments */
                 const jerry_api_length_t args_cnt) /** < number of function arguments */
 {
-  if (args_cnt > 0
+  if (args_cnt == 1
       && args_p[0].type == JERRY_API_DATA_TYPE_BOOLEAN
-      && args_p[0].v_bool != true)
+      && args_p[0].v_bool == true)
+  {
+    return true;
+  }
+  else
   {
     JERRY_ERROR_MSG ("Script assertion failed\n");
     exit (JERRY_STANDALONE_EXIT_CODE_FAIL);
   }
-
-  return true;
 } /* assert_handler */
 
+/**
+ * Provide the 'print' implementation for the engine.
+ *
+ * @return true
+ */
+static bool
+print_handler (const jerry_api_object_t *function_obj_p __attr_unused___, /** < function object */
+               const jerry_api_value_t *this_p __attr_unused___, /** < this arg */
+               jerry_api_value_t *ret_val_p __attr_unused___, /** < return argument */
+               const jerry_api_value_t args_p[], /** < function arguments */
+               const jerry_api_length_t args_cnt) /** < number of function arguments */
+{
+  for (jerry_api_length_t i = 0; i < args_cnt; i++)
+  {
+    switch (args_p[i].type)
+    {
+      case JERRY_API_DATA_TYPE_UNDEFINED:
+      {
+        printf ("undefined");
+        break;
+      }
+      case JERRY_API_DATA_TYPE_NULL:
+      {
+        printf ("null");
+        break;
+      }
+      case JERRY_API_DATA_TYPE_BOOLEAN:
+      {
+        printf ("%s", args_p[i].v_bool ? "true" : "false");
+        break;
+      }
+      case JERRY_API_DATA_TYPE_FLOAT32:
+      {
+        printf ("%f", args_p[i].v_float32);
+        break;
+      }
+      case JERRY_API_DATA_TYPE_FLOAT64:
+      {
+        printf ("%lf", args_p[i].v_float64);
+        break;
+      }
+      case JERRY_API_DATA_TYPE_STRING:
+      {
+        /*
+         * TODO:
+         *      Support print of unicode strings
+         */
+        constexpr size_t str_buffer_size_for_chars = 1024;
+        jerry_api_char_t str_buffer[str_buffer_size_for_chars + 1 /* for null character */];
+        ssize_t sz = jerry_api_string_to_char_buffer (args_p[i].v_string, str_buffer, str_buffer_size_for_chars);
+        if (sz < 0)
+        {
+          printf ("'<<<string is too long for current <print> implementation>>>'");
+        }
+        else
+        {
+          str_buffer[sz] = '\0';
+          printf ("%s", str_buffer);
+        }
+        break;
+      }
+      case JERRY_API_DATA_TYPE_OBJECT:
+      {
+        printf ("[Object]");
+        break;
+      }
+      default:
+      {
+        printf ("'<<<argument type is unsupported in current <print> implementation>>>'");
```

Comment:
Well at least I would really like to be able to print out integers. Maybe we can convert everything to int and print out that way for now, and add a FIXME entry for it. Or is it possible to somehow do a toString call on the arguments? Maybe for that the print should be a built-in method (just for easier implementation)?